rest/rest_collection.h

```diff
@@ -108,6 +108,8 @@ getDefaultDescription(RestCollectionChildEvent<Key, Value> * = 0);
 /** Base class for background tasks. */
 
 struct BackgroundTaskBase {
+    /// _initializing and _executing are transient states while 
+    /// _cancelled, _finished and _error are final states 
     enum State {_initializing, _executing, _cancelled, _finished, _error};
```

Comment:
We use CAPS for enum constants.  leading underscores are reserved for the OS and compiler.  Please rename to allcaps.